.github/scripts/calculate_file_test_rating.py

```diff
@@ -0,0 +1,123 @@
+import json
+from collections import defaultdict
+from pathlib import Path
+
+from utils import query_rockset
+
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+
+FAILED_TESTS_QUERY = """
+SELECT
+    distinct REPLACE(t.invoking_file, '.', '/') as invoking_file,
+    t.name,
+    t.classname,
+    t.file,
+    j.head_sha,
+FROM
+    commons.failed_tests_run t
+    join workflow_job j on t.job_id = j.id
+where
+    t.file is not null
+"""
+
+MERGE_BASES_QUERY = """
+select * from merge_bases
+"""
+
+
+def filter_tests(failed_tests, merge_bases):
```

Comment:
mind adding a comment explaining what this test is trying to do?

It seems like you're trying to exclude tests which were already failing on the merge base